### ISSUE TITLE
`typeof this` => `this` throw "`Cannot find name 'this'.(2304)`" for no reason

### ISSUE THREAD

**Issue by Malix-Off** [microsoft/TypeScript#54167]
# Bug Report

### 🔎 Search Terms

this, typeof this, Cannot find name 'this', 2304

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBAYwDYEMDOa4EE4G8BQAkCHALxwCMRKAFCAFxwwCeYwEAZkwBYCWaAOhABKRgDcIvACZ4AvvgD0CuCrgA9VZpVqdGgMIoAdoYjwOvQzMMoAtsDgByGHzQOBNAEwBmAAwAWYUVlVQAZAH0IyKjIuAADEFi4bnQ4MCgINigmVnsLDmAoYEMEe0NgMQLnYCRgDHx8eXwECEM0eBQyODKAd2waYQFaCkCgA)

### 💻 Code

```ts
export class A {
	x = 1
	a(x: typeof this.x): void {}
//        ^         ^^^^ Cannot find name 'this'.(2304)
//        L_____________ `x` has proper type inference nevertheless

}

const a = new A().a(1)
```

### 🙁 Actual behavior

Throw a wrong error 

### 🙂 Expected behavior

To not throw this error

**Comment by MartinJohns** [comment]
`typeof` is used to get the type of values, but `this` is no value in that position, it's a type. You want to write `this["x"]` instead.

**Comment by somebody1234** [comment]
@MartinJohns note that it doesn't error if you remove the last line though

**Comment by whzx5byb** [comment]
Note that this is a regression between 4.8.4 and 4.9.5 (and also 4.7.4 and 4.8.4).

4.7.4:
Always throw error.

4.8.4:
No error. 

4.9.5:
Throw error only with the last line.
